quartz/components/scripts/search.inline.ts

```diff
@@ -64,13 +70,60 @@ function highlight(searchTerm: string, text: string, trim?: boolean) {
       }
       return tok
     })
+    .slice(startIndex, endIndex + 1)
     .join(" ")
 
   return `${startIndex === 0 ? "" : "..."}${slice}${
     endIndex === tokenizedText.length - 1 ? "" : "..."
   }`
 }
 
+function highlightHTML(searchTerm: string, el: HTMLElement) {
+  // try to highlight longest tokens first
+  const p = new DOMParser()
+  const tokenizedTerms = tokenizeTerm(searchTerm)
+  const html = p.parseFromString(el.innerHTML, "text/html")
+
+  const createHighlightSpan = (text: string) => {
+    const span = document.createElement("span")
+    span.className = "highlight"
+    span.textContent = text
+    return span
+  }
+
+  const highlightTextNodes = (node: Node) => {
+    if (node.nodeType === Node.TEXT_NODE) {
+      let nodeText = node.nodeValue || ""
+      tokenizedTerms.forEach((term) => {
+        const escapedTerm = term.replace(/[.*+?^${}()|[\]\\]/g, "\\$&") // Escape special characters
+        const regex = new RegExp(escapedTerm, "gi")
+        if (regex.test(nodeText)) {
+          const matches = nodeText.match(regex)
```

Comment:
this is redundant, match will only return true if it matches

Reply:
Updated. I have removed this.